File: SelectSort.java
public class SelectSort implements Sorting
{
	@Override
	public void sort(int[] tab) 
	{
		int [] liczby = tab.clone(); 
		int temp, min_index;
		
		for(int i=0; i<liczby.length-1; i++)
		{
			min_index = i;
			
			for(int j=i+1; j<liczby.length; j++)
			{
				if(liczby[j] < liczby[min_index])
				{
					min_index = j;
				}
			}
			temp = liczby[i];
			liczby[i]=liczby[min_index];
			liczby[min_index] = temp;
		}
		
	}

}

File: selectSort.py

def selectSort(inputTab):
    for i in range(len(inputTab)):
        minIndex = i
        for j in range(i,len(inputTab)):
            if inputTab[minIndex] > inputTab[j]:
                minIndex = j
        if minIndex != i:
            inputTab[i], inputTab[minIndex] = inputTab[minIndex], inputTab[i]


